### ISSUE TITLE
CalendarDay UI is buggy when using european date formats

### ISSUE THREAD

**Issue by dimitriaatos** [keystonejs/keystone#2751]
When using a format that puts the day before the month e.g.
```javascript
const { CalendarDay } = require('@keystonejs/fields');

keystone.createList('Date', {
  fields: {
    date: {
      type: CalendarDay,
      format: 'DD/MM/YYYY', <---
      yearRangeFrom: 1901,
      yearRangeTo: 2018,
    },
  },
});
```
and typing a date in the CalendarDay UI, CalendarDay tries to guess if the day or the month comes first and it defaults to month first if the input is ambiguous. For example, typing "15/3/2020" and focusing out from the input field will generate "15/03/2020", but typing "2/3/2020" will generate "03/02/2020". This behaviour becomes particularly buggy if the user focuses in and out of the field again, in that case the already formatted date "03/02/2020" is parsed and formatted again and date and month switch positions again.
![DateTimeBug](https://user-images.githubusercontent.com/16505191/79360766-07e6d880-7f4d-11ea-97f7-d6b6ed942449.gif)

**Comment by Vultraz** [comment]
The date-changes-on-focus bug was a long-standing issue. It should be fixed with `@keystonejs/field` 9.0.1 (see #2424)

**Comment by dimitriaatos** [comment]
My bad, I falsely assumed that `@keystonejs/fields` and the keystone repo where updated at the same time, so I didn't check any commits done before the day I installed `@keystonejs/fields` (3 days ago).
